audio-worklet/src/noisegate-demo.js

```diff
@@ -0,0 +1,284 @@
+    this.noisegateGain_.gain.setValueAtTime(1, t);
+    this.withoutNoisegateGain_.gain.setValueAtTime(0, t);
+
+    if (this.playing) {
+      this.attackSlider_.enable();
+      this.releaseSlider_.enable();
+      this.thresholdSlider_.enable();
+    }
+  }
+
+  withoutNoisegateSelected() {
+    const t = this.context_.currentTime + 0.01;
+    this.withoutNoisegateGain_.gain.setValueAtTime(1, t);
+    this.noisegateGain_.gain.setValueAtTime(0, t);
+
+    this.attackSlider_.disable();
+    this.releaseSlider_.disable();
+    this.thresholdSlider_.disable();
+  }
+
+  /**
+   * Change the threshold.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value The new threshold.
+   */
+  setThreshold(value) {
+    // TODO: When the param-controller update lands from the synthesizer branch,
+    // the stringToNumber conversion in this and subsequent methods should be
+    // removed.
+    const numericValue = parseInt(value);
+    if (numericValue < -100) throw 'The minimum threshold rate is -100.';
+    if (numericValue > 0) throw 'The maximum threshold rate is 0.';
+    this.noisegateScriptProcessor_.threshold = numericValue;
+  }
+
+  /**
+   * Change attack.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value The new attack value.
+   */
+  setAttack(value) {
+    const numericValue = parseFloat(value);
+    if (numericValue < 0) throw 'The minimum attack value is 0.';
+    this.noisegateScriptProcessor_.attack = numericValue;
+  }
+
+  /**
+   * Change release.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value The new sample rate reduction.
+   */
+  setRelease(value) {
+    const numericValue = parseInt(value);
+    if (numericValue < 0) throw 'The minimum reduction rate is 0.';
+    this.noisegateScriptProcessor_.release = numericValue;
+  }
+
+  /**
+   * Change the volume.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value The new gain.
+   */
+  setGain(value) {
+    this.masterGain_.gain.value = parseFloat(value);
+  }
+
+  /**
+   * Change the volume of the noise source.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value The new gain.
+   */
+  setNoiseGain(value) {
+    this.noiseGain_.gain.value = parseFloat(value);
+  }
+
+  /**
+   * Start audio processing and configure UI elements.
+   */
+  start() {
+    this.noiseSource_ =
+        new AudioBufferSourceNode(this.context_, {buffer: this.noiseBuffer_});
+    this.speechSource_ =
+        new AudioBufferSourceNode(this.context_, {buffer: this.speechBuffer_});
+    
+    this.noiseSource_.connect(this.noiseGain_);
+    this.speechSource_.connect(this.speechGain_);
+
+    this.speechSource_.onended = () => {
+      this.sourceButton_.enable();
+      this.attackSlider_.disable();
+      this.releaseSlider_.disable();
+      this.thresholdSlider_.disable();
+    }
```

Comment:
"a very low signal to noise ratio" - what do you mean by this?

There's nothing to it. If you install one via Package Control, it will just nag you when it finds nits like this in your code.